test/feature/fake-cloud-tasks-feature.js

```diff
@@ -0,0 +1,212 @@
+            headers: { correlationId: "some-epic-id" },
+          },
+        },
+      });
+    });
+
+    And("the task has been run", async () => {
+      await fakeCloudTasks.processMessages();
+    });
+
+    Then("we should have received the task name", () => {
+      response.should.eql([ { name: "test-task" } ]);
+    });
+
+    And("we should have recorded the message", () => {
+      fakeCloudTasks.recordedMessages()[0].should.eql({
+        queue: config.cloudTasks.queue,
+        httpMethod: "post",
+        message: { type: "something", id: "some-id" },
+        url: "/foo/bar",
+        headers: { correlationId: "some-epic-id" },
+        correlationId: "some-epic-id",
+      });
+    });
+
+    And("we should have recorded a message handler response", () => {
+      fakeCloudTasks
+        .recordedMessageHandlerResponses()[0]
+        .should.deep.include({ body: { status: "ok" }, statusCode: 200 });
+    });
+  });
+
+  Scenario("successfully create a task with a GET request", () => {
+    Given("we have enabled publish", async () => {
+      const app = express();
+      app.use(express.json());
+      app.get("/foo/bar", (req, res) => res.status(200).json({ status: "ok" }).send());
+
+      await fakeCloudTasks.enablePublish(app);
+    });
+
+    let cloudTask;
+    And("we have a cloud task instance", () => {
+      cloudTask = new CloudTasksClient();
+    });
+
+    let response;
+    When("creating a task", async () => {
+      response = await cloudTask.createTask({
+        parent: config.cloudTasks.queue,
+        task: {
+          httpRequest: {
+            url: `${config.cloudTasks.selfUrl}/foo/bar`,
+            httpMethod: "get",
+            headers: { correlationId: "some-epic-id" },
+          },
+        },
+      });
+    });
+
+    And("the task has been run", async () => {
+      await fakeCloudTasks.processMessages();
+    });
+
+    Then("we should have received the task name", () => {
+      response.should.eql([ { name: "test-task" } ]);
+    });
+
+    And("we should have recorded the message", () => {
+      fakeCloudTasks.recordedMessages()[0].should.eql({
+        queue: config.cloudTasks.queue,
+        httpMethod: "get",
+        url: "/foo/bar",
+        headers: { correlationId: "some-epic-id" },
+        correlationId: "some-epic-id",
+      });
+    });
+
+    And("we should have recorded a message handler response", () => {
+      fakeCloudTasks
+        .recordedMessageHandlerResponses()[0]
+        .should.deep.include({ body: { status: "ok" }, statusCode: 200 });
+    });
+  });
+
+  Scenario("trying to run sequence without enabling publication", () => {
+    let error;
+    When("trying to process messages", async () => {
+      try {
+        await fakeCloudTasks.processMessages();
+      } catch (err) {
+        error = err;
+      }
+    });
+
+    Then("we should should have a processed sequence", () => {
```

Comment:
`we should should have a processed sequence` -> `we should have received an error`